select debugger gem dependent on ruby version

cruwe opened the pull request:
pry-debugger depends on debugger, which cannot be installed on
ruby>=2.0. An extensive, lengthy and sometimes unnerving discussion can
be found on https://github.com/cldwalker/debugger/issues/125

On May 16, 2014, cldwalker suggested moving to byebug for
rubies>=2. This is reflected in the change to Gemfile.

	modified:   Gemfile

Thanks for your work on puppet-datacat! Cheers!

cruwe committed:
select debugger gem dependent on ruby version

pry-debugger depends on debugger, which cannot be installed on
ruby>=2.0. An extensive, lengthy and sometimes unerving discussion can
be found on https://github.com/cldwalker/debugger/issues/125

On May 16, 2014, cldwalker suggested moving to byebug for
rubies>=2. This is reflected in the change to Gemfile.

	modified:   Gemfile

richardc commented:
This is blowing up some of the travis builds, as their bundler doesn't know what `:mri_22` is.

Probably better to gate on `RUBY_VERSION` rather than use this platforms option, as it doesn't cope with expressions in the negative.

cruwe committed:
select debugger gem dependent on ruby version - 2nd try

pry-debugger depends on debugger, which cannot be installed on
ruby>=2.0. An extensive, lengthy and sometimes unerving discussion can
be found on https://github.com/cldwalker/debugger/issues/125

On May 16, 2014, cldwalker suggested moving to byebug for
rubies>=2. This is reflected in the change to Gemfile.

As it turns out, travisci has problems working with the mri platform
directive, so bacd7829ee2ce86c3242c7aa45c987cdb3fafcde has been
reworked to resort to simple if-else logic.

cruwe commented:
I am sorry to cause this hassle. I was under the impression that travis would deal with gem to ruby requirments differently. Apparently, sometimes builds fail choking on gems requiring ruby > 1.9.3 and I do not want to mess with your test setup. Interestingly,  139.9 got through, but I lack the travis expertise to really understand the issue. Please excuse the noise ...

richardc commented:
It looks like you've hit some natural bitrot/bit shifting as `json_pure` (a transient dependency from `puppet`) has dropped support for ruby < 2.x too.  I'll get that fixed up on master and then you can rebase your changes over mine.

richardc commented:
I fixed up the rake and json_pure issues in #40 so if you can rebase this branch over master we should be testing just your changes (though there is no ruby 2.x in the travis matrix, so it might be worth extending that too)

mxftw commented:
@richardc can you merge this PR into master?

richardc commented:
I'm waiting on @cruwe rebasing, as per the preceding comment.

richardc commented:
Stomped over this with #41 so closing

cruwe commented:
Oh gosh ... it totally escaped my notice that I had a task there. So sorry! In any case, thanks for your work!
